pkg/baseconn/conn_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package baseconn
+
+import (
+	"errors"
+	"testing"
+
+	tcontext "github.com/pingcap/dm/pkg/context"
+	"github.com/pingcap/dm/pkg/retry"
+	"github.com/pingcap/dm/pkg/terror"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	. "github.com/pingcap/check"
+)
+
+func TestSuite(t *testing.T) {
+	TestingT(t)
+}
+
+var _ = Suite(&testBaseConnSuite{})
+
+type testBaseConnSuite struct {
+}
+
+func (t *testBaseConnSuite) TestBaseConn(c *C) {
+	baseConn, err := NewBaseConn("error dsn", nil)
+	c.Assert(err.(*terror.Error).Code(), Equals, terror.ErrCode(10001))
```

Comment:
```suggestion
	c.Assert(terror.ErrDBDriverError.Equal(err), IsTrue)
```
we can check with specific `*Error` directly, and the same with following error check

Reply:
Done